sql/logictest: segfault when running `make testlogic` -  regression

On my system (freebsd) running `make testlogic` fails with a segfault:

```
go list -tags 'make x86_64_unknown_freebsd12.0 make x86_64_unknown_freebsd12.0' -f \
'go test -v  -tags '\''make x86_64_unknown_freebsd12.0 make x86_64_unknown_freebsd12.0'\'' -ldflags '\''-X github.com/cockroachdb/cockroach/pkg/build.typ=development -X github.com/cockroachdb/cockroach/pkg/build.typ=development'\'' -i -c {{.ImportPath}} -o {{.Dir}}/{{.Name}}.test' ./pkg/sql/logictest | \
/usr/local/bin/bash
github.com/cockroachdb/cockroach/pkg/sql/logictest
testmain
gmake[1]: Leaving directory '/data/home/kena/src/go/src/github.com/cockroachdb/cockroach'
cd pkg/sql/logictest && ./logictest.test -test.run "Test(CCL)?Logic//^show_source$/" -test.timeout 4m -test.v -show-sql
unexpected fault address 0x426ad60
fatal error: fault
[signal SIGSEGV: segmentation violation code=0x1 addr=0x426ad60 pc=0x543844]

goroutine 1 [running, locked to thread]:
runtime.throw(0x1e55c27, 0x5)
        /data/home/kena/src/go1.9/src/runtime/panic.go:605 +0x95 fp=0xc420248318 sp=0xc4202482f8 pc=0x513aa5
runtime.sigpanic()
        /data/home/kena/src/go1.9/src/runtime/signal_unix.go:374 +0x227 fp=0xc420248368 sp=0xc420248318 pc=0x52a657
runtime.memeqbody()
        /data/home/kena/src/go1.9/src/runtime/asm_amd64.s:1460 +0x114 fp=0xc420248370 sp=0xc420248368 pc=0x543844
strings.HasPrefix(...)
        /data/home/kena/src/go1.9/src/strings/strings.go:495
github.com/cockroachdb/cockroach/pkg/build.IsRelease(...)
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/build/info.go:57
github.com/cockroachdb/cockroach/pkg/util/log.glob..func2(0x1e80025, 0x22)
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:201 +0xb5 fp=0xc4202483b0 sp=0xc420248370 pc=0xb68755
github.com/cockroachdb/cockroach/pkg/util/log.init()
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:205 +0x2c3 fp=0xc420248408 sp=0xc4202483b0 pc=0xb69d53
github.com/cockroachdb/cockroach/pkg/util/protoutil.init()
        <autogenerated>:1 +0x96 fp=0xc420248450 sp=0xc420248408 pc=0xb73586
github.com/cockroachdb/cockroach/pkg/roachpb.init()
        <autogenerated>:1 +0x73 fp=0xc420248538 sp=0xc420248450 pc=0xcb51c3
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.init()
        <autogenerated>:1 +0xf0 fp=0xc42024bf28 sp=0xc420248538 pc=0xde4b30
github.com/cockroachdb/cockroach/pkg/sql/logictest.init()
        <autogenerated>:1 +0xb4 fp=0xc42024bf70 sp=0xc42024bf28 pc=0x15b0184
main.init()
        <autogenerated>:1 +0x4e fp=0xc42024bf80 sp=0xc42024bf70 pc=0x15b29ee
runtime.main()
        /data/home/kena/src/go1.9/src/runtime/proc.go:173 +0x1c5 fp=0xc42024bfe0 sp=0xc42024bf80 pc=0x515135
runtime.goexit()
        /data/home/kena/src/go1.9/src/runtime/asm_amd64.s:2337 +0x1 fp=0xc42024bfe8 sp=0xc42024bfe0 pc=0x5441b1
gmake: *** [Makefile:769: testlogic] Error 2

Compilation exited abnormally with code 2 at Thu Dec 28 18:29:43
```

I bisected this back to 1ece2d2d02d519fa5685469c7f237494fe4f55fe.

Concretely, the following command always generates a valid `logictest.test` which works:

```
go test  -tags ' make x86_64_unknown_freebsd12.0' -ldflags ' \
   -X github.com/cockroachdb/cockroach/pkg/build.typ=development' \
   -c -o bin/logictest.test ./pkg/sql/logictest
```

Whereas the following always creates a `logictest.test` what fails with a segfault:

```
go list -tags 'make x86_64_unknown_freebsd12.0 make x86_64_unknown_freebsd12.0' -f \
  'go test -v  -tags '\''make x86_64_unknown_freebsd12.0 make x86_64_unknown_freebsd12.0'\'' \
     -ldflags '\''-X github.com/cockroachdb/cockroach/pkg/build.typ=development \
                      -X github.com/cockroachdb/cockroach/pkg/build.typ=development'\'' \
     -i -c {{.ImportPath}} -o {{.Dir}}/{{.Name}}.test' \
   ./pkg/sql/logictest | /usr/local/bin/bash
```

--- a/Makefile
+++ b/Makefile
@@ -748,7 +748,7 @@ testrace: TESTTIMEOUT := $(RACETIMEOUT)
 FIND_RELEVANT := find $(PKG_ROOT) -name node_modules -prune -o
 
 pkg/%.test: main.go $(shell $(FIND_RELEVANT) ! -name 'zcgo_flags.go' -name '*.go')
-	$(MAKE) testbuild GOFLAGS='$(GOFLAGS)' TAGS='$(TAGS)' LINKFLAGS='$(LINKFLAGS)' PKG='./pkg/$(*D)'
+	$(MAKE) testbuild PKG='./pkg/$(*D)'
 
 bench: ## Run benchmarks.
 bench: TESTS := -

